applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -1122,6 +1317,67 @@ namespace Kratos
     //************************************************************************************
     //************************************************************************************
 
+    void BaseSolidElement::CalculateAndAddKm(
```

Comment:
ok, the keys to the optimization is to ensure to always access row by row.

so
Cij += Btrans_ik D_kl B_lj = B_ki D_kl B_lj

can be implemented as 
for k=0 ...
    for l = 0...
        for i = 0...
             double tmp = B_ki*D_lk
            for j = 0...
                  C(i,j) += tmp*B_lj
                
where (if i did it correctly) every access is by row